src/main/java/uk/gov/mca/beacons/api/gateways/LegacyBeaconGatewayImpl.java

```diff
@@ -57,4 +69,54 @@ public void deleteAll() {
       .findById(id)
       .map(legacyBeaconMapper::fromJpaEntity);
   }
+
+  @Override
+  public List<LegacyBeacon> findAllByHexIdAndEmail(String hexId, String email) {
+    final String sql =
+      "SELECT " +
+      "id, hex_id, owner_email, use_activities, owner_name, created_date, last_modified_date, beacon_status, data FROM legacy_beacon WHERE owner_email = ?" +
+      " AND hex_id = ?";
+
+    List<LegacyBeaconEntity> legacyBeaconEntities = jdbcTemplate.query(
+      sql,
+      preparedStatement -> {
+        preparedStatement.setString(1, email);
+        preparedStatement.setString(2, hexId);
+      },
+      this::mapRow
+    );
+
+    return legacyBeaconEntities
+      .stream()
+      .map(legacyBeaconMapper::fromJpaEntity)
+      .collect(Collectors.toList());
+  }
+
+  private LegacyBeaconEntity mapRow(ResultSet resultSet, int rowNum) {
+    LegacyBeaconEntity legacyBeaconEntity = new LegacyBeaconEntity();
+
+    try {
+      legacyBeaconEntity.setId(UUID.fromString(resultSet.getString("id")));
+      legacyBeaconEntity.setHexId(resultSet.getString("hex_id"));
+      legacyBeaconEntity.setOwnerEmail(resultSet.getString("owner_email"));
+      legacyBeaconEntity.setUseActivities(
+        resultSet.getString("use_activities")
+      );
+      legacyBeaconEntity.setOwnerName(resultSet.getString("owner_name"));
+      legacyBeaconEntity.setData(dataColumnToMap(resultSet.getString("data")));
+    } catch (Exception e) {
+      log.error(String.valueOf(e));
+    }
```

Comment:
Need to work out why the `mapRow` method isn't able to catch the Jackson exceptions here.